R/mfx-tidiers.R

```diff
@@ -0,0 +1,241 @@
+
+#' @rdname tidy.mfx
+#' @method tidy poissonmfx
+tidy.poissonmfx <- tidy.mfx
+
+#' @rdname tidy.mfx
+#' @method tidy probitmfx
+tidy.probitmfx <- tidy.mfx
+
+
+
+#' @templateVar class mfx
+#' @template title_desc_augment
+#' 
+#' @inherit tidy.mfx params examples
+#' 
+#' @param x A `betamfx`, `logitmfx`, `negbinmfx`, `poissonmfx`, or `probitmfx`  object.
+#' @template param_data
+#' @template param_newdata
+#' @param type.predict Passed to [stats::predict.glm()] `type`
+#'   argument. Defaults to `"link"`.
+#' @param type.residuals Passed to [stats::residuals.glm()] and
+#'   to [stats::rstandard.glm()] `type` arguments. Defaults to `"deviance"`.
+#' @template param_se_fit
+#' @template param_unused_dots
+#' 
+#' @evalRd return_augment(
+#'   ".se.fit",
+#'   ".hat",
+#'   ".sigma",
+#'   ".std.resid",
+#'   ".cooksd"
+#' )
+#'
+#' @details This generic augment method wraps [augment.glm()] for objects from
+#'   the `mfx` package.
+#'   
+#' @family mfx tidiers
+#' @seealso [augment()], [augment.glm()], [mfx::betamfx()], [mfx::logitmfx()], [mfx::negbinmfx()], [mfx::poissonmfx()], [mfx::probitmfx()]
+#' @export
+augment.mfx <- function(x, 
+                        data = model.frame(x$fit),
+                        newdata = NULL,
+                        type.predict = c("link", "response", "terms"),
+                        type.residuals = c("deviance", "pearson"),
+                        se_fit = FALSE, ...) {
+  ## Use augment.glm() method on internal fit object
+  df <- augment.glm(x$fit, 
+                data = data,
+                newdata = newdata,
+                type.predict = type.predict,
+                type.residuals = type.residuals,
+                se_fit = se_fit, ...)
+  
+  df
+}
+
+#' @rdname augment.mfx
+#' @method augment betamfx
+augment.betamfx <- augment.mfx
+
+#' @rdname augment.mfx
+#' @method augment logitmfx
+augment.logitmfx <- augment.mfx
+
+#' @rdname augment.mfx
+#' @method augment negbinmfx
+augment.negbinmfx <- augment.mfx
+
+#' @rdname augment.mfx
+#' @method augment poissonmfx
+augment.poissonmfx <- augment.mfx
+
+#' @rdname augment.mfx
+#' @method augment probitmfx
+augment.probitmfx <- augment.mfx
+
+
+
+#' @templateVar class mfx
+#' @template title_desc_glance
+#' 
+#' @inherit tidy.mfx params examples
+#' 
+#' @param x A `betamfx`, `logitmfx`, `negbinmfx`, `poissonmfx`, or `probitmfx`  object.
+#' @template param_unused_dots
+#'
+#' @evalRd return_glance(
+#'   "null.deviance",
+#'   "df.null",
+#'   "logLik",
+#'   "AIC",
+#'   "BIC",
+#'   "deviance",
+#'   "df.residual",
+#'   "nobs"
+#' )
+#'
+#' @details This generic glance method wraps [glance.glm()] for objects from
+#'   the `mfx` package.   
+#'
+#' @family mfx tidiers
+#' @seealso [glance()], [glance.glm()], [mfx::betamfx()], [mfx::logitmfx()], [mfx::negbinmfx()], [mfx::poissonmfx()], [mfx::probitmfx()]
+#' @export
+glance.mfx <- function(x, ...) {
+  ## Use glance.glm() method on internal fit object
+  ret <- glance(x$fit)
```

Comment:
Let's be really explicit here and go with:

```suggestion
  ret <- glance.glm(x$fit)
```

I think it makes it more clear this is a shim and this if it breaks, we'll just need to write a new `glance.mfx` tidier

Reply:
Agreed. Similar to your comment above, I guess we might as well drop the intermediate `ret` object too since there's no real point in saving it.